### PR TITLE
Feature/add metadata to environment filters

### PR DESCRIPTION
**Pull request by bomoko**
This PR introduces a small extension to the project/environmentsByFactSearch queries, allowing both to search on project metadata.

The queries now support the following

```query environmentbymetadata {
  environmentsByFactSearch(input: {
    filterConnective: AND
    filters: [
      {lhsTarget: PROJECT_METADATA, name: "exampleKey", contains: "exampleValue" }
    ]
  }) {
		environments {
      id
      name
    }
  }
}
```

It introduces a `lhsTarget` of `PROJECT_METADATA` and, as with the other target types, allows one to specify a key and value for the search. The above will match any environments with the metadata key/value pair "exampleKey:exampleValue".
This can, further, be combined with the existing filters.

### CODE REVIEW

File: services/api/src/resources/fact/resolvers.ts [hunk 1]
```diff
@@ -525,6 +527,7 @@ const buildConditionsForFactSearchQuery = (filterDetails: any, factQuery: any, p
           innerBuilder = innerBuilder.orWhere(builderFactory(filter, i));
         }
       });
+      console.log(innerBuilder.toString());
```
Comment (timclifford) [inline services/api/src/resources/fact/resolvers.ts]: console log

Comment (bomoko) [reply services/api/src/resources/fact/resolvers.ts]: fixed :)

**Comment by tobybellwood** [comment]
@bomoko - should we just close this if we're looking at re-doing metadata, or is it ok to merge now and modify later?

**Comment by tobybellwood** [comment]
Closing pending metadata rewrite

**Pull request closed without merging**